packages/core/src/setup.ts

```diff
@@ -145,6 +146,49 @@ export function setup<
 } & {
   [K in RequiredSetupKeys<TChildrenMap>]: unknown;
 }): {
+  /**
+   * Creates a state config that is strongly typed. This state config can be
+   * used to create a machine.
+   *
+   * @example
+   *
+   * ```ts
+   * const lightMachineSetup = setup({
+   *   // ...
+   * });
+   *
+   * const green = lightMachineSetup.createStateConfig({
+   *   on: {
+   *     timer: {
+   *       actions: 'doSomething'
+   *     }
+   *   }
+   * });
+   *
+   * const machine = lightMachineSetup.createMachine({
+   *   initial: 'green',
+   *   states: {
+   *     green,
+   *     yellow,
+   *     red
+   *   }
+   * });
+   * ```
+   */
+  createStateConfig: (
```

Comment:
don't we want to make this generic so the returned `config` would be of a strict/strong shape? with that `state.matches` and stuff could continue to produce specific types